ros2topic/ros2topic/verb/pub.py

```diff
@@ -131,17 +137,26 @@ def main(self, *, args):
 
 
 def main(args):
-
+    qos_profile = get_pub_qos_profile()
     qos_profile_name = args.qos_profile
-    if not qos_profile_name:
-        qos_profile_name = default_profile_str
-
-    qos_profile = qos_profile_from_short_keys(
-        qos_profile_name, reliability=args.qos_reliability, durability=args.qos_durability,
-        depth=args.qos_depth, history=args.qos_history)
+    if qos_profile_name:
+        qos_profile = rclpy.qos.QoSPresetProfiles.get_from_short_key(qos_profile_name)
+    if args.qos_history:
+        qos_profile.history = QoSHistoryPolicy.get_from_short_key(args.qos_history)
+    if args.qos_durability:
+        qos_profile.durability = QoSDurabilityPolicy.get_from_short_key(args.qos_durability)
+    if args.qos_reliability:
+        qos_profile.reliability = QoSReliabilityPolicy.get_from_short_key(args.qos_reliability)
+    if args.qos_depth and args.qos_depth >= 0:
+        qos_profile.depth = args.qos_depth
+    else:
+        if (qos_profile.durability == rclpy.qos.QoSDurabilityPolicy.TRANSIENT_LOCAL
+                and qos_profile.depth == 0):
+            qos_profile.depth = 1
```

Comment:
instead of duplicating most of the logic from `qos_profile_from_short_keys()`, we could modify it so the first argument can either be a `QoSProfile` object or a profile name (e.g. checking isinstance(qos_profile, str) at the beggining).

Reply:
Thanks for your review. I agree on the inconvenience of duplicating logic from qos_profile_from_short_keys(), but I'm not so sure about using isinstance. So I took a slightly different approach. PTAL

Reply:
> I'm not so sure about using isinstance

duck typing is common practice in python.

> So I took a slightly different approach. PTAL

LGTM!